src/term_ctl.c

```diff
@@ -17,64 +17,203 @@
 
 #include "term_ctl.h"
 
+#ifdef _WIN32
+#include <io.h>
+#include <limits.h>
+#include <windows.h>
+
+#ifndef STDOUT_FILENO
+#define STDOUT_FILENO   1
+#endif
+
+#ifndef STDERR_FILENO
+#define STDERR_FILENO   2
+#endif
+
+static CONSOLE_SCREEN_BUFFER_INFO console_info;
+static BOOL                       console_redirected = FALSE;
+static HANDLE                     console_hnd = NULL;
+static FILE                      *console_file = NULL;
```

Comment:
Can't we pass this as parameter?

Reply:
Were? It's just a copy of `fp` from `term_init()` and `term_has_color()`. Probably `stdout`.

Reply:
Just pass fp as parameter then and remove console_file here?

Reply:
I don't understand. What about `_term_exit()`? An `atexit()` function is `void`.

Reply:
A global `struct r_cfg cfg` with some added terminal stuff then?

Reply:
The global r_cfg is nearly gone. It won't be available anymore. But we can pass a void *ctx to every term_ call.

Reply:
I'll patch that later if you just merge this PR. Then I have something to edit. (I'm still a *git n00b* and won't be able to edit this PR now).